### PR TITLE
Phase2-hgx294Z Modify the getCorner algo of HGCalGeometry to take care of rotated layers

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Modify the getCorner algo of HGCalGeometry to take care of rotated layers

#### PR validation:

Use cfg's in Geometry/HGCalGeometry/test area as we'll as runTheMatrix test workflows

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

**Comment by bsunanda** [comment]
@AdrianoDee, @srimanob please review and sign

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 1]
```diff
@@ -613,6 +613,30 @@ unsigned int HGCalDDDConstants::layersInit(bool reco) const {
   return (reco ? hgpar_->depthIndex_.size() : hgpar_->layerIndex_.size());
 }
 
+std::pair<float, float> HGCalDDDConstants::localToGlobal8(
+    int lay, int waferU, int waferV, double localX, double localY, bool reco, bool debug) const {
+  double x(localX), y(localY);
+  bool rotx =
+      ((!hgpar_->layerType_.empty()) && (hgpar_->layerType_[lay - hgpar_->firstLayer_] == HGCalTypes::WaferCenterR));
+#ifdef EDM_ML_DEBUG
+  if (debug)
```
Comment (srimanob) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc:622]: Why do you need this `if` under `ifdef`?

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 2]
```diff
@@ -613,6 +613,30 @@ unsigned int HGCalDDDConstants::layersInit(bool reco) const {
...
+#endif
+  if (!reco) {
+    x *= HGCalParameters::k_ScaleToDDD;
+    y *= HGCalParameters::k_ScaleToDDD;
+  }
+  const auto& xy = waferPositionNoRot(lay, waferU, waferV, reco, debug);
+  x += xy.first;
+  y += xy.second;
+#ifdef EDM_ML_DEBUG
+  if (debug)
```
Comment (srimanob) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc:634]: Same here.

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by srimanob** [comment]
+Upgrade

**Comment by civanch** [comment]
+1

was approved by geometry before

**Comment by qliphy** [comment]
+1

**Pull request merged**